[Bug]: o3r create fails with preset all

### Package name

create

### Package version

12.0.0-rc.0

### Reproduction steps

`npm create @o3r@~12.0.0-rc test-preset-all -- --preset=all`

Confirm all the prompts with default value.

For "Do you want to setup Playwright test framework for E2E?" you can choose yes or no, it doesn't make any difference.


### Current result

```shell
A merge conflicted on path "/jest.config.js".
npm ERR! code 3
```

### Expected result

Create command is successful. 

### Additional comments

The fix should be confirmed by a integration test in `o3r/create`.

--- a/packages/@o3r/create/README.md
+++ b/packages/@o3r/create/README.md
@@ -56,3 +56,4 @@ On top of them, the following options can be provided to the initializer:
 - `--yarn`: Enforce `yarn` package manager. This option will be ignored if `--package-manager` is already specified.
 - `--yarn-version`: specify the version of yarn to use (default: `latest`)
 - `--exact-o3r-version`: use a pinned version for [Otter packages](https://github.com/AmadeusITGroup/otter/blob/main/docs/README.md).
+- `--preset <preset_name>`: the collection of Otter modules to install. If not provided it defaults to `basic`. The full list of presets is available in [@o3r/core](https://www.npmjs.com/package/@o3r/core#presets)

--- a/packages/@o3r/create/src/index.it.spec.ts
+++ b/packages/@o3r/create/src/index.it.spec.ts
@@ -50,6 +50,64 @@ describe('Create new otter project command', () => {
     expect(() => packageManagerRunOnProject(appName, true, { script: 'build' }, execInAppOptions)).not.toThrow();
   });
 
+  test('should generate a project with preset all', async () => {
+    const { workspacePath, packageManagerConfig, o3rVersion } = o3rEnvironment.testEnvironment;
+    const inProjectPath = path.join(workspacePath, workspaceProjectName);
+    const execWorkspaceOptions = { ...defaultExecOptions, cwd: workspacePath };
+    const execInAppOptions = { ...defaultExecOptions, cwd: inProjectPath };
+    const createOptions = [
+      '--package-manager', getPackageManager(),
+      '--skip-confirmation',
+      '--preset all',
+      ...(packageManagerConfig.yarnVersion ? ['--yarn-version', packageManagerConfig.yarnVersion] : [])];
+
+    // TODO: remove it when fixing #1356
+    await fs.mkdir(inProjectPath, { recursive: true });
+    setPackagerManagerConfig(packageManagerConfig, execInAppOptions);
+
+    expect(() => packageManagerCreate({ script: `@o3r@${o3rVersion}`, args: [workspaceProjectName, ...createOptions] }, execWorkspaceOptions, 'npm')).not.toThrow();
+    expect(existsSync(path.join(inProjectPath, 'angular.json'))).toBe(true);
+    expect(existsSync(path.join(inProjectPath, 'package.json'))).toBe(true);
+    expect(() => packageManagerInstall(execInAppOptions)).not.toThrow();
+
+    const appName = 'test-application';
+    const inApplicationPath = path.join(inProjectPath, 'apps', appName);
+    expect(() => packageManagerExec({ script: 'ng', args: ['g', 'application', appName] }, execInAppOptions)).not.toThrow();
+    expect(existsSync(path.join(inProjectPath, 'project'))).toBe(false);
+    expect(existsSync(path.join(inApplicationPath, 'package.json'))).toBe(true);
+    expect(existsSync(path.join(inApplicationPath, 'tsconfig.json'))).toBe(true);
+    expect(() => packageManagerRunOnProject(appName, true, { script: 'build' }, execInAppOptions)).not.toThrow();
+  });
+
+  test('should generate a project with preset cms', async () => {
+    const { workspacePath, packageManagerConfig, o3rVersion } = o3rEnvironment.testEnvironment;
+    const inProjectPath = path.join(workspacePath, workspaceProjectName);
+    const execWorkspaceOptions = { ...defaultExecOptions, cwd: workspacePath };
+    const execInAppOptions = { ...defaultExecOptions, cwd: inProjectPath };
+    const createOptions = [
+      '--package-manager', getPackageManager(),
+      '--skip-confirmation',
+      '--preset cms',
+      ...(packageManagerConfig.yarnVersion ? ['--yarn-version', packageManagerConfig.yarnVersion] : [])];
+
+    // TODO: remove it when fixing #1356
+    await fs.mkdir(inProjectPath, { recursive: true });
+    setPackagerManagerConfig(packageManagerConfig, execInAppOptions);
+
+    expect(() => packageManagerCreate({ script: `@o3r@${o3rVersion}`, args: [workspaceProjectName, ...createOptions] }, execWorkspaceOptions, 'npm')).not.toThrow();
+    expect(existsSync(path.join(inProjectPath, 'angular.json'))).toBe(true);
+    expect(existsSync(path.join(inProjectPath, 'package.json'))).toBe(true);
+    expect(() => packageManagerInstall(execInAppOptions)).not.toThrow();
+
+    const appName = 'test-application';
+    const inApplicationPath = path.join(inProjectPath, 'apps', appName);
+    expect(() => packageManagerExec({ script: 'ng', args: ['g', 'application', appName] }, execInAppOptions)).not.toThrow();
+    expect(existsSync(path.join(inProjectPath, 'project'))).toBe(false);
+    expect(existsSync(path.join(inApplicationPath, 'package.json'))).toBe(true);
+    expect(existsSync(path.join(inApplicationPath, 'tsconfig.json'))).toBe(true);
+    expect(() => packageManagerRunOnProject(appName, true, { script: 'build' }, execInAppOptions)).not.toThrow();
+  });
+
   test('should generate a project with a library', async () => {
     const { workspacePath, packageManagerConfig, o3rVersion } = o3rEnvironment.testEnvironment;
     const inProjectPath = path.join(workspacePath, workspaceProjectName);

--- a/packages/@o3r/testing/schematics/ng-add/index.ts
+++ b/packages/@o3r/testing/schematics/ng-add/index.ts
@@ -147,33 +147,38 @@ function ngAddFn(options: NgAddSchematicsSchema): Rule {
         if (workingDirectory === undefined) {
           throw new O3rCliError(`Could not find working directory for project ${options.projectName || ''}`);
         } else {
-          const packageJsonFile = tree.readJson(`${workingDirectory}/package.json`) as PackageJson;
-          packageJsonFile.scripts ||= {};
-          packageJsonFile.scripts.test = 'jest';
-          tree.overwrite(`${workingDirectory}/package.json`, JSON.stringify(packageJsonFile, null, 2));
-          const rootRelativePath = path.posix.relative(workingDirectory, tree.root.path.replace(/^\//, './'));
-          const jestConfigFilesForProject = () => mergeWith(apply(url('./templates/project'), [
-            template({
-              ...options,
-              rootRelativePath,
-              isAngularSetup: tree.exists('/angular.json')
-            }),
-            move(workingDirectory),
-            renameTemplateFiles()
-          ]), MergeStrategy.Overwrite);
-
-          const jestConfigFilesForWorkspace = () => mergeWith(apply(url('./templates/workspace'), [
-            template({
-              ...options,
-              tsconfigPath: `./${['tsconfig.base.json', 'tsconfig.json'].find((tsconfigBase) => tree.exists(`./${tsconfigBase}`))}`
-            }),
-            move(tree.root.path),
-            renameTemplateFiles()
-          ]), MergeStrategy.Default);
-          rules.push(
-            jestConfigFilesForProject,
-            jestConfigFilesForWorkspace
-          );
+          if (workspaceProject) {
+            const packageJsonFile = tree.readJson(`${workingDirectory}/package.json`) as PackageJson;
+            packageJsonFile.scripts ||= {};
+            packageJsonFile.scripts.test = 'jest';
+            tree.overwrite(`${workingDirectory}/package.json`, JSON.stringify(packageJsonFile, null, 2));
+            const rootRelativePath = path.posix.relative(workingDirectory, tree.root.path.replace(/^\//, './'));
+            const jestConfigFilesForProject = () => mergeWith(apply(url('./templates/project'), [
+              template({
+                ...options,
+                rootRelativePath,
+                isAngularSetup: tree.exists('/angular.json')
+              }),
+              move(workingDirectory),
+              renameTemplateFiles()
+            ]), MergeStrategy.Overwrite);
+            rules.push(jestConfigFilesForProject);
+          }
+          if (tree.exists('/jest.config.js')) {
+            context.logger.info('Jest configuration files already exist at the root of the project.');
+          } else {
+            const jestConfigFilesForWorkspace = () => mergeWith(apply(url('./templates/workspace'), [
+              template({
+                ...options,
+                tsconfigPath: `./${['tsconfig.base.json', 'tsconfig.json'].find((tsconfigBase) => tree.exists(`./${tsconfigBase}`))}`
+              }),
+              move(tree.root.path),
+              renameTemplateFiles()
+            ]), MergeStrategy.Default);
+            rules.push(
+              jestConfigFilesForWorkspace
+            );
+          }
         }
       }
 

